runtime/src/main/scala/akka/grpc/javadsl/Metadata.scala

```diff
@@ -49,3 +48,15 @@ trait Metadata {
    */
   def asScala: scaladsl.Metadata
 }
+
+/**
+ * Not for user extension
+ */
+@ApiMayChange
+@DoNotInherit
+trait RichMetadata extends Metadata {
```

Comment:
Would it make sense to use a postfix instead for easier discoverability with code completion?
`MetadataExtended`, `MetadataEnriched`

Reply:
Good idea